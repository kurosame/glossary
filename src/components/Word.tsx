import CodeHighlighter from '@/components/CodeHighlighter'
import type { WordState } from '@/modules/word'
import { Card, CardContent, CardHeader, Chip } from '@material-ui/core'
import React from 'react'
import ReactMarkdown from 'react-markdown/with-html'
import styled from 'styled-components'

interface Props {
  word: WordState
}

const TitlesSpan = styled.span`
  margin-right: 5px;
`

const Word: React.FC<Props> = p => (
  <Card>
    <CardHeader title={<h3>{p.word.id}</h3>} data-testid="card-id" />
    <CardContent data-testid="card-titles">
      {p.word.titles.map(t => (
        <TitlesSpan key={t}>
          <Chip label={t} />
        </TitlesSpan>
      ))}
    </CardContent>
    <CardContent data-testid="card-description">
      <ReactMarkdown source={p.word.description} renderers={{ code: CodeHighlighter }} escapeHtml={false} />
    </CardContent>
  </Card>
)

export default Word
