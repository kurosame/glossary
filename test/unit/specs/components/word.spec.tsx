import Word from '@/components/Word'
import { mount } from 'enzyme'
import React from 'react'

const word = {
  id: 'React',
  titles: ['React', 'React.js'],
  description: 'It a \nReact'
}

const wrapper = mount(<Word word={word} />)

test('Data binding from the word.id to CardHeader.title', () => {
  expect(
    wrapper
      .find('[data-test="id"]')
      .first()
      .text()
  ).toEqual('React')
})

test('Data binding from the word.titles to CardContent', () => {
  expect(
    wrapper
      .find('[data-test="titles"]')
      .first()
      .text()
  ).toEqual('ReactReact.js')
})

test('Data binding from the word.description to CardContent', () => {
  expect(
    wrapper
      .find('[data-test="description"]')
      .first()
      .text()
  ).toEqual('It a \nReact')
})

test('Match the snapshot', () => {
  expect(wrapper.html()).toMatchSnapshot()
})
