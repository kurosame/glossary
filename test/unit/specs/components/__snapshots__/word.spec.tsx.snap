// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the snapshot 1`] = `
"<div class=\\"MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1\\"><div class=\\"MuiCardHeader-root-29\\" data-test=\\"id\\"><div class=\\"MuiCardHeader-content-32\\"><span class=\\"MuiTypography-root-35 MuiTypography-headline-40 MuiCardHeader-title-33\\"><b>React</b></span></div></div><div class=\\"MuiCardContent-root-59\\" data-test=\\"titles\\"><div role=\\"button\\" class=\\"MuiChip-root-60\\" tabindex=\\"-1\\"><span class=\\"MuiChip-label-65\\">React</span></div><div role=\\"button\\" class=\\"MuiChip-root-60\\" tabindex=\\"-1\\"><span class=\\"MuiChip-label-65\\">React.js</span></div></div><div class=\\"MuiCardContent-root-59\\" data-test=\\"description\\"><div>It a 
React</div></div></div>"
`;
